index.d.ts

```diff
@@ -0,0 +1,221 @@
+// Type definitions for redux-little-router 15.0.0
+// Project: https://github.com/FormidableLabs/redux-little-router
+// Definitions by: priecint <https://github.com/priecint>
+//                 parkerziegler <https://github.com/parkerziegler>
+// TypeScript version: 2.4
+
+import * as React from "react";
+import { Action, Reducer, Middleware, StoreEnhancer } from "redux";
+
+export type Query = { [key: string]: string };
+export type Params = { [key: string]: string };
+
+export interface Routes {
+  [index: string]: {
+    [index: string]: RouteDefinition;
+  };
+}
+
+export interface RouteDefinition {
+  [index: string]: string | RouteDefinition;
+}
+
+export type LocationOptions = {
+  persistQuery?: boolean;
+  updateRoutes?: boolean;
+};
+
+export interface HistoryLocation {
+  hash?: string,
+  key?: string
+  pathname: string,
+  search?: string,
+  state?: {},
+}
+
+export interface Location extends HistoryLocation {
+  basename?: string;
+  options?: LocationOptions;
+  params?: Params;
+  previous?: Location;
+  query?: Query;
+  queue?: Array<Location>;
+  result?: {};
+  routes?: {};
+}
+
+export interface State {
+  router: Location;
+}
+
+export type Href = string | Location;
+
+export const LOCATION_CHANGED: string;
+export const PUSH: string;
+export const REPLACE: string;
+export const GO: string;
+export const GO_BACK: string;
+export const GO_FORWARD: string;
+export const POP: string;
+export const REPLACE_ROUTES: string;
+export const DID_REPLACE_ROUTES: string;
+
+export function push(href: Href, options: LocationOptions): {
+  type: string;
+  payload: Location;
+}
+
+export function replace(href: Href, options: LocationOptions): {
+  type: string;
+  payload: Location;
+}
+
+export function go(index: number): {
+  type: string;
+  payload: number;
+};
+
+export function goBack(): {
+  type: string;
+}
+
+export function goForward(): {
+  type: string;
+}
+
+export function locationDidChange(location: Location): {
+  type: string;
+  payload: Location;
+};
+
+export function initializeCurrentLocation(location: Location): {
+  type: string;
+  payload: Location;
+};
+
+export function replaceRoutes(routes: {}): {
```

Comment:
TS noob question, can the `routes` conform to the `Routes` interface above?

Reply:
Good catch, we should be able to do that!

Reply:
@tptee I'll get on this later tonight and plan to merge tomorrow. Need to do some introspection around [index signatures](https://basarat.gitbooks.io/typescript/docs/types/index-signatures.html). I'll ping you when things are ready!